shell/platform/android/io/flutter/embedding/engine/deferredcomponents/PlayStoreDeferredComponentManager.java

```diff
@@ -53,6 +55,8 @@
   private @NonNull SparseArray<String> sessionIdToState;
   private @NonNull Map<String, Integer> nameToSessionId;
 
+  protected @NonNull Map<Integer, List<String>> loadingUnitIdToModuleNames;
```

Comment:
Shouldn't this be `Map<Integer, String>`? How can a loading unit to be contained in multiple modules?

Reply:
It is possible to include the same loading unit in more than one module. For example, if you had a credit card feature, and wanted to include it in multiple locales but not china, then you could include the loading unit in multiple modules. This would just pick the first module that includes it, which isn't optimal, but a full system that can analyze which modules to pick with the same loading unit is quite a bit of work. This is an edge case in general, but adding the multiple modules here allows better expansion later.

Reply:
In that case the "credit card" feature should be its own module, and when user is using it from China we should just avoid using them?

I don't think we should conflate the two things together though, feature modules and locale.

Otherwise what could happen is that when the user is in US, and when the code tries to do `credit_card.loadLibrary()`, it could then download the loading unit from feature module EU, which contains a lot of other things that the user won't need.

Reply:
Restricting a loading unit to be only assigned to one module seems like a reasonable restriction for now. In any case, removing this restriction in the future would be safe and non-breaking, so I can make this just take a single module name for now.